### DISCUSSION TITLE
What's the best model for extracting instrumental only?

### DISCUSSION THREAD

**Discussion by codebespawler** [Anjok07/ultimatevocalremovergui#444] [Q&A] [answered]
What's the best model for extracting instrumental only?

**Comment by thecatontheceiling** [comment, marked as answer]
MDX-NET UVR-MDX-NET Inst 3 but if you want to get highest quality result you should try using ensemble mode instead (combine results of multiple models together)

**Reply by codebespawler** [reply]
Thank you!

**Reply by xzuyn** [reply]
which models, and what ensemble mode do you recommend?

**Reply by thecatontheceiling** [reply]
my current ensemble mode for inst/vocal is Kim_Vocal_1, MDX-NET Inst 3, MDX-NET Inst Main and htdemucs_ft. you can fiddle with the specific settings of the models by right clicking on an empty space and choosing "advanced options", if you don't know what everything means you can also enable "help hints" somewhere in the settings to make it easier to understand.

**Reply by xzuyn** [reply]
> my current ensemble mode for inst/vocal is Kim_Vocal_1, MDX-NET Inst 3, MDX-NET Inst Main and htdemucs_ft. you can fiddle with the specific settings of the models by right clicking on an empty space and choosing "advanced options", if you don't know what everything means you can also enable "help hints" somewhere in the settings to make it easier to understand.

thanks

**Reply by xzuyn** [reply]
> my current ensemble mode for inst/vocal is Kim_Vocal_1, MDX-NET Inst 3, MDX-NET Inst Main and htdemucs_ft. you can fiddle with the specific settings of the models by right clicking on an empty space and choosing "advanced options", if you don't know what everything means you can also enable "help hints" somewhere in the settings to make it easier to understand.

what ensemble algorithm do you use? the default Max Spec/Min Spec, or the new Average/Average?

**Reply by thecatontheceiling** [reply]
depends, if you want to get a more detailed result choose Max Spec, and if you want the "cleanest" result (fewer artifacts and bleed-ins basically) choose Min Spec. if you're not sure whichever you want you can use Average.

(for example: Vocal/Instrumental option in stem output with Max Spec/Min Spec selected means that the Vocals will be ensembled with Max and the instrumental will be ensembled by Min.)

**Reply by gionaguidi** [reply]
Hi there,

sorry for my stupid question but I am new here and trying to figure out how it works. When you list the modes you use in your ensemble workflow, you mean that you process the output with a new mode every time, or did you get to apply multiple modes at the same time ?

thanks

**Reply by thecatontheceiling** [reply]
> Hi there,
> 
> sorry for my stupid question but I am new here and trying to figure out how it works. When you list the modes you use in your ensemble workflow, you mean that you process the output with a new mode every time, or did you get to apply multiple modes at the same time ?
> 
> thanks 

there's an "ensemble mode" and you can select multiple models to run your song through. nowadays I strong suggest you to use the new MDX23C model and place it first in the ensemble (you can download it in the Download Center in UVR)

**Reply by nachoruiz** [reply]
Can someone explain to me how the order of the models is established, if it can be established?

**Reply by Gyramuur** [reply]
And what other models do you use in the ensemble? @sabaasa

**Reply by thecatontheceiling** [reply]
> And what other models do you use in the ensemble? @sabaasa 

I'm not using that model anymore

I'm using the mdx23c models, specifically MDX23C-8KFFT-InstVoc_HQ. I forgot what it is called in the UVR download center or if it's a VIP model or not.

**Reply by Gyramuur** [reply]
Can't seem to find it anywhere, might be subscription only. Do any models support separating vocals/instrumental to stereo tracks? Seems like everything I use crunches the tracks down to mono. Or is that a setting somewhere I can change?

**Reply by thecatontheceiling** [reply]
> Can't seem to find it anywhere, might be subscription only. Do any models support separating vocals/instrumental to stereo tracks? Seems like everything I use crunches the tracks down to mono. Or is that a setting somewhere I can change?

personally I started doing everything on Google colab instead of pc, because it's faster.
I use this one:

https://colab.research.google.com/github/jarredou/MVSEP-MDX23-Colab_v2/blob/v2.3/MVSep-MDX23-Colab.ipynb

best settings for this (in most cases) is the following:

bigShifts to 11

overlap_instvoc to 8
overlap_vitlarge to 8

use voc_ft checked (takes a while to process, you might want to disable this if you don't care about instrumentals bleeding in vocal)
weight_vocft to 1

leave the rest at the defaults, enjoy. I got very good results from this :) you will need to drop the files in your Google drive and then right click your files in colab and "copy as path" and paste it in the "input" box.

**Reply by COOLak** [reply]
Hi, I followed your guide, it took like 25 minutes for a 25-minute WAV file, then apparently it finished, but there's no output file. Do you know what could be the reason? Thanks!

Here's the info from the Google Colabs console:

`Input audio: (2, 64928651) Sample rate: 44100
Processing vocals with VitLarge model...
 55% 6/11 [20:15<16:54, 202.88s/it]^C`
 
 So apparently it finished at 55%? Weird. But below it says: "Took 24 minutes 17 seconds. Finished at xx:xx." (where xx:xx is the exact time when it finished). But the output folder remains empty. Even after the second attempt.

**Reply by thecatontheceiling** [reply]
I'm not sure if the free google colab machine can handle 25 minute files lol, i've only ever used it for 3 minute songs

**Reply by COOLak** [reply]
Do you know for sure that it will finish properly if I buy some credits? I can't figure if it means uninterrupted work or it can still throw a captcha mid-process. Or I should buy a VM from them and work on it. The latter is too technical for me though, haha. On the other hand, don't wanna throw money in vain.

**Reply by thecatontheceiling** [reply]
id suggest just using UVR then :) I think if you input the VIP code in the model download center there's a variant of the MDX23C model there to download.

**Reply by COOLak** [reply]
Oh, I know, but it takes way too long even on my $6500 Macbook Pro M2 Max. I'd rather figure out how to use it on Google Colabs :)

**Reply by Jay987654321** [reply]
Please suggest models to only extract music from the track no drums, no vocals

**Reply by Shashank-Singh1729** [reply]
How many more models I need to download if I use "Ensemble Mode" ?

**Comment by codplay89** [comment]
For the cleanest possible results for now that pretty much removes all of the vocals with little to no vocal bleed, I use MDX-NET: UVR MDX-NET INST 3 + UVR-MDX-NET Inst Main + 7_HP2-UVR +Demucs: v4 | htdemucs_ft - Shifts: 2 - Ens. Algo.: Avg/Avg Denoise: Off
I have instrumentals on nhmusicinstrumentals on YouTube. the most recent instrumental I made with this formula is NF trust feat tech nine

**Reply by it2prog** [reply]
Is this a good setting for having a clean vocal? Have seen some using Denoise: On for vocals.

**Reply by cris-pin** [reply]
> For the cleanest possible results for now that pretty much removes all of the vocals with little to no vocal bleed, I use MDX-NET: UVR MDX-NET INST 3 + UVR-MDX-NET Inst Main + 7_HP2-UVR +Demucs: v4 | htdemucs_ft - Shifts: 2 - Ens. Algo.: Avg/Avg Denoise: Off I have instrumentals on nhmusicinstrumentals on YouTube. the most recent instrumental I made with this formula is NF trust feat tech nine

Like yr youtube! Would it be possible for you to paste in a screenshot of the GUI with these options configured in at all please man? The GUI is a bit counter-intuitive .. or it could just be me ;) Thanks

**Reply by NeonPro66** [reply]
How do you change Denoise? Im new to this :)

**Reply by danthekidd** [reply]
Hey there. Go on Ensemble mode and right click any blank spot on the UI (besides in the title bar). After that, click on Advanced MDX-Net Settings. Uncheck Denoise Output if it is checked and close. I hope I helped you 🤞

**Reply by emilfro** [reply]
How do i choose all those models? I can find UVR MDX-NET INST 3 and Demucs: v4 | htdemucs_ft. But how do i choose the others?

Is it inside the secondary model settings where i can choose a model for vocals, other, bass and drums?

**Reply by Shaung6688** [reply]
which type of your Laptop level?? RTX4090??

**Reply by Gamegami49** [reply]
Mate I am on a 5070 and it took 8 hours to filter a 4 minute track, what are you on

**Reply by JaziGuj** [reply]
Damn...I am on a core i5 and intel uhd graphics and a 4 min track takes around 3 mins. What are you doing?

**Reply by kojinmalia** [reply]
Somebody please tell me what "Shifts: 2" means here?

**Comment by GitBoy-Star** [comment]
im doing voice overs of a certain show and i want to extract the best instrumental part as possible! im having trouble finding the best ones do u have any ideas by chance?

**Comment by Roopchand2023** [comment]
I just need to separate the main instrument of the instrumental track. ( for instance, the lead instruments like guitar, or accordian or harmonica or piano from the instrumental track. Please suggest the Model/s and the processing method. Thank you.

**Comment by gionaguidi** [comment]
Thank you very much for your explanation. I will try to figure out where in the app i can create this ensemble processing chain
Il 11 ott 2023, 04:13 +0200, Lyssa ***@***.***>, ha scritto:

**Reply by duet-music** [reply]
Hi everybody,
I'm new here. Just wanted to ask if somebody tried to remove a solo instrument from instrumental pieces? I'm intested in removing violin in particular. Thank you in advance!

**Comment by sanbarcam1** [comment]
Este programa é simplismente fantastico, o melhor de todos os tempos, parabens ao criador...

**Comment by Gyramuur** [comment]
Personally I'd say just download UVR 5 and run it locally; even if you
don't have a good GPU it can run in CPU mode, just takes a little longer.

**Comment by Prem-2024** [comment]
ive used the settings recommded, but my instrumental track is muffled.. any ideas what i can do?

**Comment by Newingland360** [comment]
If anyone's interested in keeping not only instrumentals but also the backup vocals/harmonies, I've used ensemble mode with VR Arc: 5_HP-Karaoke-UVR and VR Arc: 6_HP-Karaoke-UVR. They were titled a little differently when downloading, but they were the only two in the VR Arch dropdown with Karaoke in the title.  There are Karaoke downloads in the MDX-Net dropdown but I haven't tried them yet.
This works pretty well at deciding and removing the main vocals while leaving the backups.  Some songs are trickier, like guy and girl trading off lead vocals, but it does pretty well overall.  There are also some occasional glitches, where vocals are not removed for the occasional riff when someone ooohs or ahhhs, but again, I get one of those every three or four songs I've tried.
Either way, I'm very happy with this combo for getting the lead vocals removed while keeping instruments and backup vocals.
I'll also say, my full system is:
Audacity (free) to record the song from whatever source plays on my computer to an mp3.
UVR5 (free) to strip out the lead vocals into separate vocal and instrumental mp3 tracks.
Amazing Slowdowner (paid, sorry) to change the pitch up/down by semitones to bring it to my range.  This was key for me, because you can finally sing a lot of songs that are originally too high or low!

**Reply by who-else123** [reply]
What's the difference between 5_HP-Karaoke-UVR and 6_HP-Karaoke-UVR? I can't find any information on them and their differences, I'm trying to extract the best possible backing vocals with them

**Reply by fred913** [reply]
Thanks DUDE IT IS ABSOLUTELY THE SHIT!!! Worked insanely great for a Chinese song. Now all i need is to gently fix some of the vocal samples in the instrumental by simply dragging them and assemble to the AI outputs. Thanks alot, really. 🤗

**Comment by obiyomida** [comment]
Here's my setup with no backup vocal.

- MDX23C-InstVocHQ
- MDXNET HQ 5 & Inst Main
- Demucs_ft

Overlap Mdx- 0.25
Conversion pitch - 0 (Lower ghosting afaik, I never went negative so 0 is fine for me)
Overlap demucs - 0.5
Ensemble Algo - Average/Average.
Denoise - off
Spectral inversion - On

I've tried using VR models but the audio seems muddy so I skipped it.
I've been using all that 4 models to all songs. You only need to experiment the settings depends on the song. There's no AIO settings so good luck.

This is not a setting for extracting each instrument.

*UPDATE*
Model
-MDX23C-InstVocHQ & Duality v2
-MDXNet - BS-Roformer-Viperx-1297

Ensemble Algo - Min/Min

-Mdx settings-
Volume compensation - 0
Overlap mdx - 0.85
Segment Size mdx - 512
Volume Compensation - 0
Spectral Inversion - Tick

**Reply by reporeporeporepo** [reply]
> Here's my setup with no backup vocal.
> 
> * MDX23C-InstVocHQ
> * MDXNET HQ 5 & Inst Main
> * Demucs_ft
> 
> Overlap Mdx- 0.25 Conversion pitch - 0 (Lower ghosting afaik, I never went negative so 0 is fine for me) Overlap demucs - 0.5 Ensemble Algo - Average/Average. Denoise - off Spectral inversion - On
> 
> I've tried using VR models but the audio seems muddy so I skipped it. I've been using all that 4 models to all songs. You only need to experiment the settings depends on the song. There's no AIO settings so good luck.
> 
> This is not a setting for extracting each instrument.

Hello man, I'm trying to follow your specs but I'm confused, it produced 4 instrumental files, is that expected? I'm expecting just one instrumental combined/ processed by 4 models.

Here's screenshot of my main window:
![image](https://github.com/user-attachments/assets/588524cc-93dc-4d65-a784-e4ebcd3a4f52)

Can you tell me what I'm doing wrong? Thank you.

**Reply by obiyomida** [reply]
> 
> Hello man, I'm trying to follow your specs but I'm confused, it produced 4 instrumental files, is that expected? I'm expecting just one instrumental combined/ processed by 4 models.

Did you mean there's 4 output ? Try disable "Save all output" in ensemble settings. If that's what you meant.

**Reply by reporeporeporepo** [reply]
> > Hello man, I'm trying to follow your specs but I'm confused, it produced 4 instrumental files, is that expected? I'm expecting just one instrumental combined/ processed by 4 models.
> 
> Did you mean there's 4 output ? Try disable "Save all output" in ensemble settings. If that's what you meant.

nice much better! thanks man! these settings actually sound good compared to other i've tried, what's your settings for acappela or vocal only im curious

**Comment by GianniBK** [comment]
Actually, for an optimal result you should use the Kim Duality Version 2 Roformer model in MDX mode, without using ensemble

**Reply by SILENT21SORA35** [reply]
the Kim Duality Version 2 Roformer model? where, there's nothing like that,

**Comment by GianniBK** [comment]
Hi there! For the RoFormer version, you need to download the new package. You can find it directly via this link: https://github.com/Anjok07/ultimatevocalremovergui/releases/download/v5.6/UVR_1_15_25_22_30_BETA_full.exe Once you have installed the new version of UVR, go to the download section and download the Dualty version 2 package.

**Reply by SILENT21SORA35** [reply]
Would it work, though? Every model in UVR5 I tried, and even other  tools like UVR5, and I still have the same issues: the vocals are muffled and bleeding into the instrument parts

**Comment by GianniBK** [comment]
Have you tried putting the separated tracks into a DAW (like FL Studio, Ableton, or Logic)?Since the AI extraction can sometimes leave the tracks sounding a bit dull, you actually need to post-process them. Try adding an EQ to boost the high frequencies (around 4kHz - 8kHz) to fix the muffled vocal sound. Then, you can use a Compressor and a Gate on the instrumental track to squash down and hide the remaining vocal bleeding.Give it a shot inside a DAW, it usually cleans up the results drastically!

**Comment by Bassizq** [comment]
I'm trying to extract instruments in different tracks, I need to make some arrangements  to a song, and I need to separate instruments, so i can edit for example drums, and have control of editing bass, vocals and harmonies. Does anyone have used a successful processing model to achieve this?
